### ISSUE TITLE
Move all generated files to a temporary directory

### ISSUE THREAD

**Issue by dietmarw** [OpenModelica/OMPython#34]
Running examples in OMPython will place all generated files in the current directory. 
This is not very nice and clutters up the workspace. OMPython should generate all files in some temporary directory instead.

**Comment by bilderbuchi** [comment]
Note that the tempfile module in the standard library can be used to create and automatically clean up temporary directories.

**Comment by sjoelund** [comment]
I disagree. I usually read the generated files such as result files for example for plotting. It would really mess up scripts if files were suddenly not generated in the same location as previously.

Could be an option when creating the object though.

⁣Sent from BlueMail ​

**Comment by dietmarw** [comment]
Being able to specify the location would probably be a good enhancement
then. E.g. Instead of a system temp into a subfolder. OMedit uses a system
temp as far as i know.

16. jun. 2017 7.45 p.m. skrev "Martin Sjölund" <notifications@github.com>:

**Comment by sjoelund** [comment]
One of the first things OMEdit does is call `cd("/tmp/OpenModelica_userName/OMEdit/")`. It's quite explicit in that.

**Comment by sjoelund** [comment]
@adeas31 you "recently" added some new flag for simulations to deal with this, right? So new you can set fileNamePrefix to generate files in one location (which ModelSystem could default to a temp directory) and use a different directory for output files generated by the simulation.

**Comment by ondras12345** [comment]
As far as I can tell, this is already implemented:
* `ModelicaSystem.__init__` has an argument called `customBuildDirectory`. If left unset, it creates a temporary directory using `tempfile.mkdtemp()`.
* `ModelicaSystem.simulate()` has an arg called `resultfile` to allow specifying the path to a custom result file. This can then be used in conjunction with `ModelicaSystem.getSolutions(resultfile=...)`.

This issue should probably be closed.
